### ISSUE TITLE
nixos-20.03 pkgs.haskellPackages.shellFor missing hoogle

### ISSUE THREAD

**Issue by Skyfold** [NixOS/nixpkgs#83783]
**Describe the bug**

The Haskell packages `shellFor` function should provide a `hoogle` executable with a database filled with the specified packages when `withHoogle = true;`. However, for the latest commit on `nixos-20.03` the `hoogle` executable is missing. For both `nixos-19.09` and `nixos-unstable` `shellFor` provides the `hoogle` executable.

**To Reproduce**

1. save this as `nixos-20.03.nix`, then call `nix-shell --pure nixos-20.03.nix`

```
{ pkgs ? import <nixpkgs> {} }:

let

  nixpkgs-src = pkgs.fetchFromGitHub {
    owner = "NixOS";
    repo = "nixpkgs";
    rev = "b0c285807d6a9f1b7562ec417c24fa1a30ecc31a";
    sha256 = "0waapr7aqz0h1fy1fqlx981ygllh91qx9sz1l2j2h59s46cdircl";
  };
  nixpkgs = import nixpkgs-src {};

  shell = nixpkgs.haskellPackages.shellFor {
    packages = ps: [ ps.rio ];
    buildInputs = [ nixpkgs.haskellPackages.cabal-install ];
    withHoogle = true;
  };

in
  shell
```

2. Type `hoogle` in the resulting shell and see "command not found"

**Expected behavior**

I would expect `shellFor` to add `hoogle` to your path when `withHoogle = true;`. Here is the same setup as above, but using `nixos-19.09`. Load this in `nix-shell --pure` and you will get `hoogle`:

```
{ pkgs ? import <nixpkgs> {} }:

let

  nixpkgs-src = pkgs.fetchFromGitHub {
    owner = "NixOS";
    repo = "nixpkgs";
    rev = "c7363c2b97e0e4eb127e6a32b3207cc00eac3409";
    sha256 = "1cn175ghzg5nzchxzcmdbwj6143q0q7am1gqdpypjdbcdkpghrqp";
  };
  nixpkgs = import nixpkgs-src {};

  shell = nixpkgs.haskellPackages.shellFor {
    packages = ps: [ ps.rio ];
    buildInputs = [ nixpkgs.haskellPackages.cabal-install ];
    withHoogle = true;
  };

in
  shell
```

This one is for `nixos-unstable`:

```
{ pkgs ? import <nixpkgs> {} }:

let

  nixpkgs-src = pkgs.fetchFromGitHub {
    owner = "NixOS";
    repo = "nixpkgs";
    rev = "ae6bdcc53584aaf20211ce1814bea97ece08a248";
    sha256 = "0hjhznns1cxgl3hww2d5si6vhy36pnm53hms9h338v6r633dcy77";
  };
  nixpkgs = import nixpkgs-src {};

  shell = nixpkgs.haskellPackages.shellFor {
    packages = ps: [ ps.rio ];
    buildInputs = [ nixpkgs.haskellPackages.cabal-install ];
    withHoogle = true;
  };

in
  shell
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
  - haskellPackages.shellFor
```

**Comment by cdepillabout** [comment]
I believe this was fixed in `master` in https://github.com/NixOS/nixpkgs/pull/80845, but was never backported to 20.03.

If you (or anyone else) would like to test this and backport it, I'll happily merge it in.

**Comment by cdepillabout** [comment]
Closed by #84181 by @schmittlauch.  Thanks!
